maven-embedder/src/test/java/org/apache/maven/cli/MavenCliTest.java

```diff
@@ -544,6 +544,19 @@ public void populatePropertiesOverwrite() throws Exception {
         assertThat(request.getUserProperties().getProperty("x"), is("false"));
     }
 
+    @Test
+    public void testPropertiesInterpolation() throws Exception {
```

Comment:
There should also be a test for the exceptional case.

Reply:
Added some tests.